### ISSUE TITLE
Error: Invalid usage of publishCreate()

### ISSUE THREAD

**Issue by neelesh-v** [balderdashy/sails#2187]
Hello Team,

I'm using publishCreate and its giving me error 

    Error: Invalid usage of publishCreate() :: Values must have an `id`, instead got ::
    { value: '540845716324a0942b226b8a' }

I just simply sibscribed model Notification and using : 
          Notification.publishCreate(notificationObj)

My model is

     module.exports = {

      schema: true,
      attributes: {

    name: {
        type: 'string',
        required: true
    },
    active: {
        type: 'string'
    },

    toJSON: function () {
     
            var obj = this.toObject();
            return obj;
        }

    }   };

I'm using sails 0.10.0-rc11, any suggestion welcome.

**Comment by bostrom** [comment]
I'm using sails ^0.11.0 and sails-mysql ~0.10.11 and face the same error `error: Invalid usage of publishCreate() :: Values must have an 'id', instead got ::` when using the blueprint POST to create a new object.
The model in question is defined as
```javascript
// api/models/thingy.js
module.exports = {
  autoPK: true,
  attributes: {
    title: {
      type: 'string',
      required: true
    },
    type: {
      type: 'integer',
      required: true
    }
  }
}
```
When I do a POST to the API with the following payload
```javascript
{
    "title": "Some title",
    "type": 2
}
```
I get the above error: `Invalid usage of publishCreate() :: Values must have an `id`, instead got ::` and the POST response contains 
```javascript
{
  "title": "Some title",
  "type": 2
}
```
i.e. the newly created object's id is missing, although it was created successfully in the database.

However, if I change the `type` attribute to, say, `thingy_type`, it works as expected:
```javascript
// api/models/thingy.js
module.exports = {
  autoPK: true,
  attributes: {
    title: {
      type: 'string',
      required: true
    },
    thingy_type: {
      type: 'integer',
      required: true,
      columnName: 'type'
    }
  }
}
```
POST: 
```javascript
{
    "title": "Some title",
    "thingy_type": 2
}
```
Response:
```javascript
{
  "id": 5,
  "title": "Some title",
  "thingy_type": 2
}
```

I'm guessing the word `type` as the attribute name is clashing with the attribute data type definition? Also, the error message is not very informative, as it sounds like there's something going on with the `id` attribute.

**Comment by CWyrtzen** [comment]
Are you still getting this error?

**Comment by sailsbot** [comment]
Thanks for posting, @neelesh-v.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
